src/cc65/wrappedcall.c

```diff
@@ -64,13 +64,13 @@ static IntPtrStack WrappedCalls;
 
 
 
-void PushWrappedCall (void *Ptr, unsigned char Val)
+void PushWrappedCall (void *Ptr, unsigned char Val, int UseBank)
 /* Push the current WrappedCall */
 {
     if (IPS_IsFull (&WrappedCalls)) {
         Error ("WrappedCall stack overflow");
     } else {
-        IPS_Push (&WrappedCalls, Val, Ptr);
+        IPS_Push (&WrappedCalls, Val | (UseBank << 8), Ptr);
```

Comment:
This seems pretty ugly to me. If you need to "encode" the new flag and the normal value into one variable here anyway I suggest to do that in the first place.

The way the code is written now, it implies that there is a value and additionally an optional flag meaning that the value needs to be interpreted in a different way.

However, this isn't true. Rather there are two types of values: The normal value and a special value. Therefore I suggest to extent the `unsigned char Val` to `unsigned Val` and set it to the value of a new macro (which is define as 0x0100) - and get rid of `UseBank` altogether.